src/create-theme-provider.js

```diff
@@ -8,10 +8,10 @@ import isObject from './is-object';
 
 export type ThemeProviderProps<Theme> = {
   children: Node,
-  theme: Theme | (outerTheme: Theme) => Theme,
+  theme: Theme | (outerTheme: Theme | null) => Theme,
 };
 
-export default function createThemeProvider<Theme>(context: Context<Theme>) {
+export default function createThemeProvider<Theme>(context: Context<Theme | null>) {
```

Comment:
Should we treat flexibly undfined and null? `Context<?Theme>`

Reply:
What I decided instead only to make our context type `null` so that when for example MaterialUI would create their context with a proper default theme it will have the correct type.

Though I made some type non-nullable.